src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java

```diff
@@ -0,0 +1,180 @@
+package com.iwaconsolti.league.manager.service.impl;
+
+import com.iwaconsolti.league.manager.model.Matches;
+import com.iwaconsolti.league.manager.model.Players;
+import com.iwaconsolti.league.manager.model.Teams;
+import com.iwaconsolti.league.manager.service.ILeagues;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service("soccerLeague")
+@Slf4j
+public class SoccerLeagueImpl implements ILeagues {
+
+    @Value("${league.teams.limit}")
+    private int teamLimit;
+
+    private List<Players> players = new ArrayList<>();
+    private List<Teams> teams = new ArrayList<>();
+    private List<Matches> matches = new ArrayList<>();
+    private int nextPlayer = 1;
+    private int nextTeam = 1;
+    private int nextMatches = 1;
+
+    @Override
+    public Players savePlayers(Players player) {
+        log.debug("Saving player {}", player);
+        player.setId(nextPlayer++);
+        players.add(player);
+        Teams team = findTeams(player.getTeamId());
+        if (team != null) {
+            if (team.getPlayers() == null) {
+                team.setPlayers(new ArrayList<>());
+            }
+            team.getPlayers().add(player);
+        } else {
+            log.warn("Team not found");
+        }
+
+        return player;
+    }
+
+    @Override
+    public Players findPlayers(int id) {
+        for (Players player : players) {
+            if (player.getId() == id) {
+                return player;
+            } else {
+                log.warn("Player not found");
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public Teams saveTeams(Teams team) {
+        if (teams.size() >= teamLimit) {
+            log.warn("Team limit exceeded");
+            return null;
+        }
+        log.debug("Saving team {}", team);
```

Comment:
Same as the other when an entry is created/updated/deleted is a good practice to log as info